minard/detector_state.py

```diff
@@ -513,15 +513,18 @@ def get_nhit_monitor(key):
 
     return dict(zip(keys,row))
 
-def get_nhit_monitor_thresholds_nearline(limit=100, offset=0, sort_by="run"):
+def get_nhit_monitor_thresholds_nearline(limit=100, offset=0, sort_by="run", run_range_low=0, run_range_high=0):
     """
     Returns a list of the latest nhit monitor records in the nearline database.
     """
     conn = engine_nl.connect()
 
-    if sort_by == "run":
+    if run_range_high:
+        result = conn.execute("SELECT * FROM nhit_monitor_thresholds WHERE run >= %s AND run <= %s "
+                              "ORDER BY run DESC, timestamp DESC LIMIT %s OFFSET %s", (run_range_low, run_range_high, limit, offset))
+    elif sort_by == "run":
         result = conn.execute("SELECT * FROM nhit_monitor_thresholds ORDER BY run DESC, timestamp DESC LIMIT %s OFFSET %s", (limit,offset))
-    if sort_by == "key":
+    elif sort_by == "key":
```

Comment:
Is this if/elif block expected to be like this? It seems like currently if you enter a high run range then you lose the ability to sort by key, but the webpage interface doesn't make it seem like these are mutually exclusive.

Reply:
Is there a way to do "ORDER BY %s", (sort_by,)? I've tried a couple things and none of them seem to work.